lib/Parse/ParseDecl.cpp

```diff
@@ -3763,6 +3792,18 @@ void Parser::ParseStructUnionBody(SourceLocation RecordLoc,
                       RecordLoc, TagDecl, FieldDecls,
                       T.getOpenLocation(), T.getCloseLocation(),
                       attrs.getList());
+  // Parse the deferred bounds expressions
+  for (const auto &Pair : deferredBoundsExpressions) {
+    EnterMemberBoundsExprRAII MemberBoundsContext(Actions);
+    DeclaratorDecl *FieldDecl = dyn_cast<DeclaratorDecl>(Pair.first);
```

Comment:
>dyn_cast [](start = 32, length = 8)

Do we need a (safer) isa check first?  Just to doubly make sure that they really are all DeclaratorDecls?